tests/Test/Xrefcheck/TrailingSlashSpec.hs

```diff
@@ -0,0 +1,45 @@
+{- SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.TrailingSlashSpec where
+
+import qualified Unsafe
+
+import Test.Hspec (Spec, describe, it, runIO)
+import Test.QuickCheck ((===))
+
+import Xrefcheck.Config
+import Xrefcheck.Core
+import Xrefcheck.Progress
+import Xrefcheck.Scan
+import Xrefcheck.Scanners.Markdown
+import Xrefcheck.Verify
+
+spec :: Spec
+spec = do
+  describe "Trailing forward slash detection" $ do
+    let config = defConfig GitHub
+    let format = specificFormatsSupport [markdownSupport (scMarkdown (cScanners config))]
+    forM_ roots $ \root -> do
+      repoInfo <- runIO $ allowRewrite False $ \rw ->
+        gatherRepoInfo rw format TraversalConfig{ tcIgnored = [] } root
+      res <- runIO $ allowRewrite False $ \rw ->
+        verifyRepo rw (cVerification config) FullMode root repoInfo
+      it ("Analysis results for the root with" <>
+          (if Unsafe.last root == '/' then "" else "out") <>
+          " the trailing forward slash should not contain the FileDoesNotExist error") $ do
+        let satisfactoryCase = maybe True (not . any checkFileDoesNotExist) $ verifyErrors res
+        satisfactoryCase === True
```

Comment:
Maybe it would be enough to check here that `repoInfo` contains something? This way the test would shorter, and would not fail when verification logic went wrong.

Reply:
I've actually tried basing my observations only on the results of `gatherRepoInfo`, but it turns out, regardless of the appearance of the trailing forward slash in the root, `gatherRepoInfo` would perform just fine! As far as I understand, such behavior is present due to the file-wise analysis being performed **before** the composition 
`Tree.zipPaths . (dirOfRoot root Tree.:/)` is called. Because of that, I thought the actual verification with `verifyRepo` was the way to go in this situation.

Reply:
(We could use [`doesFileExist`](https://hackage.haskell.org/package/directory-1.3.7.0/docs/System-Directory.html#v:doesFileExist) for this)

Reply:
Yeah, that's it, given the correct `root` where files did indeed exist, `Map FilePath FileInfo` would be non-empty, and the very problem here lied within the keys of that map, i.e. the `FilePath`s! I think I understand now, `verifyRepo` is indeed a redundant operation, and the only thing needed was to check whether these `FilePath`s do or do not exist.